ui/src/app/components/popups.tsx

```diff
@@ -123,7 +127,38 @@ export const computePopupPosition = ({side, viewport, target, element, gap}) =>
       ['top', () => position.top < 0 ? 'bottom' : 'top'],
       ['bottom', () => position.top + element.height >= viewport.height ? 'top' : 'bottom'],
       ['left', () => position.left < 0 ? 'right' : 'left'],
-      ['right', () => position.left + element.width >= viewport.width ? 'left' : 'right']
+      ['right', () => position.left + element.width >= viewport.width ? 'left' : 'right'],
+      ['bottom-left', () => position.top + element.height >= viewport.height
+          ? position.left < 0
```

Comment:
Same comment as above, would prefer either a functional approach or just a standard callback function with if/else rather than the nested ternaries

Reply:
oh boy nested switchCases!